sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -472,9 +472,12 @@ object TypeCoercion {
         // RHS is the subquery output.
         val rhs = sub.output
 
-        val commonTypes = lhs.zip(rhs).flatMap { case (l, r) =>
-          findCommonTypeForBinaryComparison(l.dataType, r.dataType, conf)
-            .orElse(findTightestCommonType(l.dataType, r.dataType))
+        val commonTypes = lhs.zip(rhs).flatMap {
+          case (l, r) if !l.dataType.isInstanceOf[DecimalType] &&
```

Comment:
can we simply call `findWiderTypeForTwo`? I think `In` and `InSubquery` should have the same type coercion policy.

Reply:
Thanks for your suggestion.